addon/mixins/route.js

```diff
@@ -412,7 +252,7 @@ const RouteMixin = Mixin.create({
       return;
     }
 
-    Ember.run.scheduleOnce('afterRender', this, 'infinityModelUpdated', {
+    run.scheduleOnce('afterRender', this, 'infinityModelUpdated', {
       lastPageLoaded: this.get('currentPage'),
       totalPages: this.get('_totalPages'),
```

Comment:
yup!  Maybe the model should use `Ember.Evented`, and the route level can subscribe to those events?  like

```
model() {
  return this.infinityModel('product')
}, 
setupController(infinityModel) {
  infinityModel.on('infinityModelUpdated', this.doSomethingAfterLoad);
}
```

This could also be a nice way to continue to support `Route#afterInfinityModel`, which I think people prefer to the route events